src/spec_procs.cpp

```diff
@@ -4375,15 +4375,17 @@ SPECIAL(desktop)
   if (obj->contains) {
     send_to_char(ch, " contains:\r\n");
     for (struct obj_data *soft = obj->contains; soft; soft = soft->next_content) {
-      if (GET_OBJ_TYPE(soft) == ITEM_DESIGN)
```

Comment:
This particular spec has been a pain in my ass for a while, so I just went ahead and rewrote it. Appreciate you doing this though-- someone having to bugfix it yet again prompted me to do the work!

Reply:
I'm good with that :)

My linux machine is my work machine so I can't put an instance of the mud on there, and my laptop is windows, so I'm using the lack of a testing environment as my excuse for not tackling more substantial re-writes.